### PR TITLE
Adding support for external authentication

### PR DESCRIPTION
**Pull request by bhoehl**
@pinepain @lstrojny
The current implementation of php-amqp is using a static value for authentication in amqp_connection_resource.c function connection_resource_constructor

amqp_rpc_reply_t res = amqp_login_with_properties(
resource->connection_state,
params->vhost,
params->channel_max,
params->frame_max,
params->heartbeat,
&custom_properties_table,
==>	AMQP_SASL_METHOD_PLAIN, <==
params->login,
params->password
);

Doing so, it is impossible to authenticate using ssl certificates, as these are requiring the sasl method AMQP_SASL_METHOD_EXTERNAL.
I have added a way to specify this method having an additional property sasl_method.
The external method enables php-amqp to authenticate using ssl, without the need of a password, instead using either the DN or CN of the certificate as the user name and by validating the certificate.
I am using this sucessfully on our system in a docker environment and would like to see this being part of the main php-ampq library.

### CODE REVIEW

**Comment by bhoehl** [comment]
@pinepain @lstrojny 
I have recreated that PR to purge those commits and addressed the issues with message naming and unit tests. Please have a look.

File: amqp_connection.c [hunk 1]
```diff
@@ -510,6 +512,14 @@ static PHP_METHOD(amqp_connection_class, __construct)
 		}
 	}
 
+	zend_update_property_long(this_ce, getThis(), ZEND_STRL("sasl_method"), INI_INT("amqp.sasl_method") TSRMLS_CC);
+
+		if (ini_arr && PHP5to7_ZEND_HASH_FIND(HASH_OF(ini_arr), "sasl_method", sizeof("sasl_method"), zdata)) {
```
Comment (pinepain) [inline amqp_connection.c]: minor formatting: Unexpected indent

File: stubs/AMQPConnection.php [hunk 2]
```diff
@@ -434,4 +434,20 @@ public function getVerify()
     public function setVerify($verify)
     {
     }
+
+    /**
+     * set authentication method
+     *
+     * @param int $method AMQP_SASL_METHOD_PLAIN | AMQP_SASL_METHOD_EXTERNAL
+     */
+    public function setSasslMethod($method)
```
Comment (pinepain) [inline stubs/AMQPConnection.php]: Double `s` in setSa**ss**lMethod()

**Comment by bhoehl** [comment]
@pinepain 
addressed

**Pull request closed without merging**